www/js/survey/enketo/service.js

```diff
@@ -139,6 +139,12 @@ angular.module('emission.survey.enketo.service', [
         // let's support both so we can label from either screen
         data.start_ts = _state.opts.trip.data.properties.start_ts;
         data.end_ts = _state.opts.trip.data.properties.end_ts;
+        // generate a UUID v4 for the survey response
+        // using method from https://stackoverflow.com/a/2117523
+        // crypto.generateUUID() is preferable, but not availble to us in the app
+        data.match_id = ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
+          (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
+        );
```

Comment:
this is fine, although given that this doesn't need to be cryptographically secure, maybe we can use a the lighter weight objectID outlined here: objectID
https://stackoverflow.com/a/66332305/4040267

This is an optimization though so am happy with deferring it until later (if ever)
You can resolve this as soon as you have read it.

Reply:
done in 586e1abc74ddbb700ed43460f2d816245f2f9b16